src/freenet/support/Waiter.java

```diff
@@ -0,0 +1,23 @@
+package freenet.support;
+
+/** Waits for some other thread to signal to proceed. */
+public class Waiter {
```

Comment:
`CountDownLatch` would be fine, too, we just need to stop rolling our own versions of stuff that is already in the JDK and has been for the last ten years. :)